Bench workflow template support

karolk91 opened the pull request:
Fixes https://github.com/polkadot-fellows/runtimes/issues/726

Introduce support for benchmark templates to Github benchmarking workflow

- [x] Does not require a CHANGELOG entry

bkontur commented on .github/workflows/runtimes-matrix.json:
```diff
@@ -16,15 +17,17 @@
     "benchmarks_exclude_extrinsics": "pallet_nomination_pools::apply_slash_fail",
-    "extra_args": "--disable-mbm-checks"
+    "extra_args": "--disable-mbm-checks",
+    "template": "benchmarking/template.hbs"
```
We have a similar setting for benchmarking `benchmarks_exclude_extrinsics`, so I would use the same syntax here:
```suggestion
    "benchmarks_xcm_template": "benchmarking/template.hbs"
```

bkontur commented on .github/scripts/cmd/cmd.py:
```diff
@@ -137,8 +137,13 @@
-            print(f'-- benchmarking {pallet} in {runtime} into {output_path}')
+            template = None
+            if pallet.startswith("pallet_xcm_benchmarks"):
+                template = config['template']
```
this condition if just for XCM weights, so I would be specific (aligned with matrix `benchmarks_xcm_template`):
```suggestion
                template = config['benchmarks_xcm_template']
```

maybe we should add to the `else` something like `template = config['template']` - but this is not necessary, we don't use template for others

bkontur commented on benchmarking/template.hbs:
```diff
@@ -0,0 +1,64 @@
+{{header}}
```
this template is dedicated for XCM weights,
so I would rename it to `xcm-bench-template.hbs`

karolk91 replied:
addresed in https://github.com/polkadot-fellows/runtimes/pull/731/commits/c7716e6b4fb9c3eab8ddfe25c5a872ad53c4fd72

bkontur commented on benchmarking/template.hbs:
```diff
@@ -0,0 +1,64 @@
+{{header}}
```
also I would better place this template under:
```
templates/xcm-bench-template.hbs
```
maybe we will have different templates for other stuff

karolk91 replied:
addresed in https://github.com/polkadot-fellows/runtimes/pull/731/commits/c7716e6b4fb9c3eab8ddfe25c5a872ad53c4fd72

karolk91 commented:
@bkontur , I have pushed a little bit different approach to allow to specify template per benchmarking pallet, wdyt ?

bkontur reviewed (approved):
Looks good, yes, it's a bit verbose considering the fact that we're using at most one template for XCM benchmarks, but theoretically we might still make use of it for storage benchmarks: https://github.com/polkadot-fellows/runtimes/issues/491.

karolk91 commented:
Since its not really possible to test this changes using comment bot (the bot will run using code from `master`), for review purposes, please find `git diff` of my local example run of

```bash
python3 .github/scripts/cmd/cmd.py bench --runtime asset-hub-polkadot --pallet pallet_xcm_benchmarks::generic pallet_xcm_benchmarks::fungible pallet_xcm
```

```diff
diff --git a/system-parachains/asset-hubs/asset-hub-polkadot/src/weights/pallet_xcm.rs b/system-parachains/asset-hubs/asset-hub-polkadot/src/weights/pallet_xcm.rs
index aa0bed738..a130fc28c 100644
--- a/system-parachains/asset-hubs/asset-hub-polkadot/src/weights/pallet_xcm.rs
+++ b/system-parachains/asset-hubs/asset-hub-polkadot/src/weights/pallet_xcm.rs
@@ -16,25 +16,26 @@

 //! Autogenerated weights for `pallet_xcm`
 //!
-//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 32.0.0
-//! DATE: 2025-01-05, STEPS: `50`, REPEAT: `20`, LOW RANGE: `[]`, HIGH RANGE: `[]`
+//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 43.0.0
+//! DATE: 2025-05-20, STEPS: `50`, REPEAT: `20`, LOW RANGE: `[]`, HIGH RANGE: `[]`
 //! WORST CASE MAP SIZE: `1000000`
-//! HOSTNAME: `ggwpez-ref-hw`, CPU: `AMD EPYC 7232P 8-Core Processor`
-//! WASM-EXECUTION: `Compiled`, CHAIN: `Some("./asset-hub-polkadot-chain-spec.json")`, DB CACHE: 1024
+//! HOSTNAME: `mac-4.home`, CPU: `<UNKNOWN>`
+//! WASM-EXECUTION: `Compiled`, CHAIN: `None`, DB CACHE: 1024

 // Executed Command:
-// ./target/production/polkadot-parachain
+// frame-omni-bencher
+// v1
 // benchmark
 // pallet
-// --chain=./asset-hub-polkadot-chain-spec.json
-// --steps=50
-// --repeat=20
-// --pallet=pallet_xcm
 // --extrinsic=*
+// --runtime=target/production/wbuild/asset-hub-polkadot-runtime/asset_hub_polkadot_runtime.wasm
+// --pallet=pallet_xcm
+// --header=/Users/kkdev/Dev/karolk91/runtimes/.github/scripts/cmd/file_header.txt
+// --output=./system-parachains/asset-hubs/asset-hub-polkadot/src/weights
 // --wasm-execution=compiled
+// --steps=50
+// --repeat=20
 // --heap-pages=4096
-// --output=./asset-hub-polkadot-weights/
-// --header=./file_header.txt

 #![cfg_attr(rustfmt, rustfmt_skip)]
 #![allow(unused_parens)]
@@ -59,14 +60,16 @@ impl<T: frame_system::Config> pallet_xcm::WeightInfo for WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `42`
 		//  Estimated: `3507`
-		// Minimum execution time: 29_860_000 picoseconds.
-		Weight::from_parts(30_671_000, 0)
+		// Minimum execution time: 15_000_000 picoseconds.
+		Weight::from_parts(16_000_000, 0)
 			.saturating_add(Weight::from_parts(0, 3507))
 			.saturating_add(T::DbWeight::get().reads(4))
 			.saturating_add(T::DbWeight::get().writes(1))
 	}
 	/// Storage: `ParachainInfo::ParachainId` (r:1 w:0)
 	/// Proof: `ParachainInfo::ParachainId` (`max_values`: Some(1), `max_size`: Some(4), added: 499, mode: `MaxEncodedLen`)
+	/// Storage: `PolkadotXcm::ShouldRecordXcm` (r:1 w:0)
+	/// Proof: `PolkadotXcm::ShouldRecordXcm` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
 	/// Storage: `ParachainSystem::UpwardDeliveryFeeFactor` (r:1 w:0)
 	/// Proof: `ParachainSystem::UpwardDeliveryFeeFactor` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
 	/// Storage: `PolkadotXcm::SupportedVersion` (r:1 w:0)
@@ -81,14 +84,16 @@ impl<T: frame_system::Config> pallet_xcm::WeightInfo for WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `74`
 		//  Estimated: `3593`
-		// Minimum execution time: 130_001_000 picoseconds.
-		Weight::from_parts(130_811_000, 0)
+		// Minimum execution time: 78_000_000 picoseconds.
+		Weight::from_parts(83_000_000, 0)
 			.saturating_add(Weight::from_parts(0, 3593))
-			.saturating_add(T::DbWeight::get().reads(6))
+			.saturating_add(T::DbWeight::get().reads(7))
 			.saturating_add(T::DbWeight::get().writes(2))
 	}
 	/// Storage: `ParachainInfo::ParachainId` (r:1 w:0)
 	/// Proof: `ParachainInfo::ParachainId` (`max_values`: Some(1), `max_size`: Some(4), added: 499, mode: `MaxEncodedLen`)
+	/// Storage: `PolkadotXcm::ShouldRecordXcm` (r:1 w:0)
+	/// Proof: `PolkadotXcm::ShouldRecordXcm` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
 	/// Storage: `System::Account` (r:2 w:2)
 	/// Proof: `System::Account` (`max_values`: None, `max_size`: Some(128), added: 2603, mode: `MaxEncodedLen`)
 	/// Storage: `XcmpQueue::DeliveryFeeFactor` (r:1 w:0)
@@ -105,14 +110,16 @@ impl<T: frame_system::Config> pallet_xcm::WeightInfo for WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `296`
 		//  Estimated: `6196`
-		// Minimum execution time: 161_501_000 picoseconds.
-		Weight::from_parts(162_771_000, 0)
+		// Minimum execution time: 102_000_000 picoseconds.
+		Weight::from_parts(108_000_000, 0)
 			.saturating_add(Weight::from_parts(0, 6196))
-			.saturating_add(T::DbWeight::get().reads(7))
+			.saturating_add(T::DbWeight::get().reads(8))
 			.saturating_add(T::DbWeight::get().writes(4))
 	}
 	/// Storage: `ParachainInfo::ParachainId` (r:1 w:0)
 	/// Proof: `ParachainInfo::ParachainId` (`max_values`: Some(1), `max_size`: Some(4), added: 499, mode: `MaxEncodedLen`)
+	/// Storage: `PolkadotXcm::ShouldRecordXcm` (r:1 w:0)
+	/// Proof: `PolkadotXcm::ShouldRecordXcm` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
 	/// Storage: `Assets::Asset` (r:1 w:1)
 	/// Proof: `Assets::Asset` (`max_values`: None, `max_size`: Some(210), added: 2685, mode: `MaxEncodedLen`)
 	/// Storage: `Assets::Account` (r:2 w:2)
@@ -129,21 +136,24 @@ impl<T: frame_system::Config> pallet_xcm::WeightInfo for WeightInfo<T> {
 	/// Proof: `ParachainSystem::PendingUpwardMessages` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
 	fn transfer_assets() -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `477`
+		//  Measured:  `514`
 		//  Estimated: `6208`
-		// Minimum execution time: 196_592_000 picoseconds.
-		Weight::from_parts(197_972_000, 0)
+		// Minimum execution time: 120_000_000 picoseconds.
+		Weight::from_parts(129_000_000, 0)
 			.saturating_add(Weight::from_parts(0, 6208))
-			.saturating_add(T::DbWeight::get().reads(10))
+			.saturating_add(T::DbWeight::get().reads(11))
 			.saturating_add(T::DbWeight::get().writes(6))
 	}
+	/// Storage: `PolkadotXcm::ShouldRecordXcm` (r:1 w:0)
+	/// Proof: `PolkadotXcm::ShouldRecordXcm` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
 	fn execute() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
-		//  Estimated: `0`
-		// Minimum execution time: 13_770_000 picoseconds.
-		Weight::from_parts(14_070_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		//  Estimated: `1485`
+		// Minimum execution time: 7_000_000 picoseconds.
+		Weight::from_parts(7_000_000, 0)
+			.saturating_add(Weight::from_parts(0, 1485))
+			.saturating_add(T::DbWeight::get().reads(1))
 	}
 	/// Storage: `PolkadotXcm::SupportedVersion` (r:0 w:1)
 	/// Proof: `PolkadotXcm::SupportedVersion` (`max_values`: None, `max_size`: None, mode: `Measured`)
@@ -151,8 +161,8 @@ impl<T: frame_system::Config> pallet_xcm::WeightInfo for WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 9_360_000 picoseconds.
-		Weight::from_parts(9_520_000, 0)
+		// Minimum execution time: 5_000_000 picoseconds.
+		Weight::from_parts(6_000_000, 0)
 			.saturating_add(Weight::from_parts(0, 0))
 			.saturating_add(T::DbWeight::get().writes(1))
 	}
@@ -160,8 +170,8 @@ impl<T: frame_system::Config> pallet_xcm::WeightInfo for WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 3_090_000 picoseconds.
-		Weight::from_parts(3_350_000, 0)
+		// Minimum execution time: 2_000_000 picoseconds.
+		Weight::from_parts(2_000_000, 0)
 			.saturating_add(Weight::from_parts(0, 0))
 	}
 	/// Storage: `PolkadotXcm::VersionNotifiers` (r:1 w:1)
@@ -182,8 +192,8 @@ impl<T: frame_system::Config> pallet_xcm::WeightInfo for WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `42`
 		//  Estimated: `3507`
-		// Minimum execution time: 37_560_000 picoseconds.
-		Weight::from_parts(38_101_000, 0)
+		// Minimum execution time: 20_000_000 picoseconds.
+		Weight::from_parts(21_000_000, 0)
 			.saturating_add(Weight::from_parts(0, 3507))
 			.saturating_add(T::DbWeight::get().reads(6))
 			.saturating_add(T::DbWeight::get().writes(4))
@@ -204,8 +214,8 @@ impl<T: frame_system::Config> pallet_xcm::WeightInfo for WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `136`
 		//  Estimated: `3601`
-		// Minimum execution time: 36_810_000 picoseconds.
-		Weight::from_parts(37_371_000, 0)
+		// Minimum execution time: 21_000_000 picoseconds.
+		Weight::from_parts(22_000_000, 0)
 			.saturating_add(Weight::from_parts(0, 3601))
 			.saturating_add(T::DbWeight::get().reads(5))
 			.saturating_add(T::DbWeight::get().writes(3))
@@ -216,45 +226,45 @@ impl<T: frame_system::Config> pallet_xcm::WeightInfo for WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 3_240_000 picoseconds.
-		Weight::from_parts(3_360_000, 0)
+		// Minimum execution time: 2_000_000 picoseconds.
+		Weight::from_parts(2_000_000, 0)
 			.saturating_add(Weight::from_parts(0, 0))
 			.saturating_add(T::DbWeight::get().writes(1))
 	}
-	/// Storage: `PolkadotXcm::SupportedVersion` (r:5 w:2)
+	/// Storage: `PolkadotXcm::SupportedVersion` (r:6 w:2)
 	/// Proof: `PolkadotXcm::SupportedVersion` (`max_values`: None, `max_size`: None, mode: `Measured`)
 	fn migrate_supported_version() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `22`
-		//  Estimated: `13387`
-		// Minimum execution time: 24_551_000 picoseconds.
-		Weight::from_parts(24_920_000, 0)
-			.saturating_add(Weight::from_parts(0, 13387))
-			.saturating_add(T::DbWeight::get().reads(5))
+		//  Estimated: `15862`
+		// Minimum execution time: 18_000_000 picoseconds.
+		Weight::from_parts(19_000_000, 0)
+			.saturating_add(Weight::from_parts(0, 15862))
+			.saturating_add(T::DbWeight::get().reads(6))
 			.saturating_add(T::DbWeight::get().writes(2))
 	}
-	/// Storage: `PolkadotXcm::VersionNotifiers` (r:5 w:2)
+	/// Storage: `PolkadotXcm::VersionNotifiers` (r:6 w:2)
 	/// Proof: `PolkadotXcm::VersionNotifiers` (`max_values`: None, `max_size`: None, mode: `Measured`)
 	fn migrate_version_notifiers() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `26`
-		//  Estimated: `13391`
-		// Minimum execution time: 24_760_000 picoseconds.
-		Weight::from_parts(25_140_000, 0)
-			.saturating_add(Weight::from_parts(0, 13391))
-			.saturating_add(T::DbWeight::get().reads(5))
+		//  Estimated: `15866`
+		// Minimum execution time: 17_000_000 picoseconds.
+		Weight::from_parts(18_000_000, 0)
+			.saturating_add(Weight::from_parts(0, 15866))
+			.saturating_add(T::DbWeight::get().reads(6))
 			.saturating_add(T::DbWeight::get().writes(2))
 	}
-	/// Storage: `PolkadotXcm::VersionNotifyTargets` (r:6 w:0)
+	/// Storage: `PolkadotXcm::VersionNotifyTargets` (r:7 w:0)
 	/// Proof: `PolkadotXcm::VersionNotifyTargets` (`max_values`: None, `max_size`: None, mode: `Measured`)
 	fn already_notified_target() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `37`
-		//  Estimated: `15877`
-		// Minimum execution time: 27_270_000 picoseconds.
-		Weight::from_parts(27_560_000, 0)
-			.saturating_add(Weight::from_parts(0, 15877))
-			.saturating_add(T::DbWeight::get().reads(6))
+		//  Estimated: `18352`
+		// Minimum execution time: 22_000_000 picoseconds.
+		Weight::from_parts(22_000_000, 0)
+			.saturating_add(Weight::from_parts(0, 18352))
+			.saturating_add(T::DbWeight::get().reads(7))
 	}
 	/// Storage: `PolkadotXcm::VersionNotifyTargets` (r:2 w:1)
 	/// Proof: `PolkadotXcm::VersionNotifyTargets` (`max_values`: None, `max_size`: None, mode: `Measured`)
@@ -270,36 +280,36 @@ impl<T: frame_system::Config> pallet_xcm::WeightInfo for WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `75`
 		//  Estimated: `6015`
-		// Minimum execution time: 36_540_000 picoseconds.
-		Weight::from_parts(37_370_000, 0)
+		// Minimum execution time: 20_000_000 picoseconds.
+		Weight::from_parts(21_000_000, 0)
 			.saturating_add(Weight::from_parts(0, 6015))
 			.saturating_add(T::DbWeight::get().reads(6))
 			.saturating_add(T::DbWeight::get().writes(2))
 	}
-	/// Storage: `PolkadotXcm::VersionNotifyTargets` (r:4 w:0)
+	/// Storage: `PolkadotXcm::VersionNotifyTargets` (r:5 w:0)
 	/// Proof: `PolkadotXcm::VersionNotifyTargets` (`max_values`: None, `max_size`: None, mode: `Measured`)
 	fn notify_target_migration_fail() -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `36`
-		//  Estimated: `10926`
-		// Minimum execution time: 17_350_000 picoseconds.
-		Weight::from_parts(17_720_000, 0)
-			.saturating_add(Weight::from_parts(0, 10926))
-			.saturating_add(T::DbWeight::get().reads(4))
+		//  Measured:  `40`
+		//  Estimated: `13405`
+		// Minimum execution time: 16_000_000 picoseconds.
+		Weight::from_parts(17_000_000, 0)
+			.saturating_add(Weight::from_parts(0, 13405))
+			.saturating_add(T::DbWeight::get().reads(5))
 	}
-	/// Storage: `PolkadotXcm::VersionNotifyTargets` (r:5 w:2)
+	/// Storage: `PolkadotXcm::VersionNotifyTargets` (r:6 w:2)
 	/// Proof: `PolkadotXcm::VersionNotifyTargets` (`max_values`: None, `max_size`: None, mode: `Measured`)
 	fn migrate_version_notify_targets() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `33`
-		//  Estimated: `13398`
-		// Minimum execution time: 24_660_000 picoseconds.
-		Weight::from_parts(25_300_000, 0)
-			.saturating_add(Weight::from_parts(0, 13398))
-			.saturating_add(T::DbWeight::get().reads(5))
+		//  Estimated: `15873`
+		// Minimum execution time: 18_000_000 picoseconds.
+		Weight::from_parts(19_000_000, 0)
+			.saturating_add(Weight::from_parts(0, 15873))
+			.saturating_add(T::DbWeight::get().reads(6))
 			.saturating_add(T::DbWeight::get().writes(2))
 	}
-	/// Storage: `PolkadotXcm::VersionNotifyTargets` (r:5 w:2)
+	/// Storage: `PolkadotXcm::VersionNotifyTargets` (r:6 w:2)
 	/// Proof: `PolkadotXcm::VersionNotifyTargets` (`max_values`: None, `max_size`: None, mode: `Measured`)
 	/// Storage: `ParachainSystem::UpwardDeliveryFeeFactor` (r:1 w:0)
 	/// Proof: `ParachainSystem::UpwardDeliveryFeeFactor` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
@@ -312,11 +322,11 @@ impl<T: frame_system::Config> pallet_xcm::WeightInfo for WeightInfo<T> {
 	fn migrate_and_notify_old_targets() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `75`
-		//  Estimated: `13440`
-		// Minimum execution time: 48_760_000 picoseconds.
-		Weight::from_parts(49_330_000, 0)
-			.saturating_add(Weight::from_parts(0, 13440))
-			.saturating_add(T::DbWeight::get().reads(9))
+		//  Estimated: `15915`
+		// Minimum execution time: 31_000_000 picoseconds.
+		Weight::from_parts(32_000_000, 0)
+			.saturating_add(Weight::from_parts(0, 15915))
+			.saturating_add(T::DbWeight::get().reads(10))
 			.saturating_add(T::DbWeight::get().writes(3))
 	}
 	/// Storage: `PolkadotXcm::QueryCounter` (r:1 w:1)
@@ -327,8 +337,8 @@ impl<T: frame_system::Config> pallet_xcm::WeightInfo for WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `1485`
-		// Minimum execution time: 3_470_000 picoseconds.
-		Weight::from_parts(3_710_000, 0)
+		// Minimum execution time: 2_000_000 picoseconds.
+		Weight::from_parts(2_000_000, 0)
 			.saturating_add(Weight::from_parts(0, 1485))
 			.saturating_add(T::DbWeight::get().reads(1))
 			.saturating_add(T::DbWeight::get().writes(2))
@@ -339,22 +349,24 @@ impl<T: frame_system::Config> pallet_xcm::WeightInfo for WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `7576`
 		//  Estimated: `11041`
-		// Minimum execution time: 33_640_000 picoseconds.
-		Weight::from_parts(34_211_000, 0)
+		// Minimum execution time: 24_000_000 picoseconds.
+		Weight::from_parts(24_000_000, 0)
 			.saturating_add(Weight::from_parts(0, 11041))
 			.saturating_add(T::DbWeight::get().reads(1))
 			.saturating_add(T::DbWeight::get().writes(1))
 	}
+	/// Storage: `PolkadotXcm::ShouldRecordXcm` (r:1 w:0)
+	/// Proof: `PolkadotXcm::ShouldRecordXcm` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
 	/// Storage: `PolkadotXcm::AssetTraps` (r:1 w:1)
 	/// Proof: `PolkadotXcm::AssetTraps` (`max_values`: None, `max_size`: None, mode: `Measured`)
 	fn claim_assets() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `23`
 		//  Estimated: `3488`
-		// Minimum execution time: 50_670_000 picoseconds.
-		Weight::from_parts(51_520_000, 0)
+		// Minimum execution time: 29_000_000 picoseconds.
+		Weight::from_parts(31_000_000, 0)
 			.saturating_add(Weight::from_parts(0, 3488))
-			.saturating_add(T::DbWeight::get().reads(1))
+			.saturating_add(T::DbWeight::get().reads(2))
 			.saturating_add(T::DbWeight::get().writes(1))
 	}
 }
diff --git a/system-parachains/asset-hubs/asset-hub-polkadot/src/weights/xcm/pallet_xcm_benchmarks_fungible.rs b/system-parachains/asset-hubs/asset-hub-polkadot/src/weights/xcm/pallet_xcm_benchmarks_fungible.rs
index ea70bf62f..e4e2cbeb5 100644
--- a/system-parachains/asset-hubs/asset-hub-polkadot/src/weights/xcm/pallet_xcm_benchmarks_fungible.rs
+++ b/system-parachains/asset-hubs/asset-hub-polkadot/src/weights/xcm/pallet_xcm_benchmarks_fungible.rs
@@ -16,30 +16,31 @@

 //! Autogenerated weights for `pallet_xcm_benchmarks::fungible`
 //!
-//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 32.0.0
-//! DATE: 2025-01-05, STEPS: `50`, REPEAT: `20`, LOW RANGE: `[]`, HIGH RANGE: `[]`
+//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 43.0.0
+//! DATE: 2025-05-20, STEPS: `50`, REPEAT: `20`, LOW RANGE: `[]`, HIGH RANGE: `[]`
 //! WORST CASE MAP SIZE: `1000000`
-//! HOSTNAME: `ggwpez-ref-hw`, CPU: `AMD EPYC 7232P 8-Core Processor`
-//! WASM-EXECUTION: `Compiled`, CHAIN: `Some("./asset-hub-polkadot-chain-spec.json")`, DB CACHE: 1024
+//! HOSTNAME: `mac-4.home`, CPU: `<UNKNOWN>`
+//! WASM-EXECUTION: Compiled, CHAIN: None, DB CACHE: 1024

 // Executed Command:
-// ./target/production/polkadot-parachain
+// frame-omni-bencher
+// v1
 // benchmark
 // pallet
-// --chain=./asset-hub-polkadot-chain-spec.json
-// --steps=50
-// --repeat=20
-// --pallet=pallet_xcm_benchmarks::fungible
 // --extrinsic=*
+// --runtime=target/production/wbuild/asset-hub-polkadot-runtime/asset_hub_polkadot_runtime.wasm
+// --pallet=pallet_xcm_benchmarks::fungible
+// --header=/Users/kkdev/Dev/karolk91/runtimes/.github/scripts/cmd/file_header.txt
+// --output=./system-parachains/asset-hubs/asset-hub-polkadot/src/weights/xcm
 // --wasm-execution=compiled
+// --steps=50
+// --repeat=20
 // --heap-pages=4096
-// --output=./asset-hub-polkadot-weights/xcm/pallet_xcm_benchmarks_fungible.rs
-// --header=./file_header.txt
+// --template=templates/xcm-bench-template.hbs

 #![cfg_attr(rustfmt, rustfmt_skip)]
 #![allow(unused_parens)]
 #![allow(unused_imports)]
-#![allow(missing_docs)]

 use frame_support::{traits::Get, weights::Weight};
 use core::marker::PhantomData;
@@ -53,9 +54,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `101`
 		//  Estimated: `3593`
-		// Minimum execution time: 42_110_000 picoseconds.
-		Weight::from_parts(42_700_000, 0)
-			.saturating_add(Weight::from_parts(0, 3593))
+		// Minimum execution time: 24_000_000 picoseconds.
+		Weight::from_parts(26_000_000, 3593)
 			.saturating_add(T::DbWeight::get().reads(1))
 			.saturating_add(T::DbWeight::get().writes(1))
 	}
@@ -65,9 +65,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `153`
 		//  Estimated: `6196`
-		// Minimum execution time: 54_461_000 picoseconds.
-		Weight::from_parts(55_000_000, 0)
-			.saturating_add(Weight::from_parts(0, 6196))
+		// Minimum execution time: 33_000_000 picoseconds.
+		Weight::from_parts(34_000_000, 6196)
 			.saturating_add(T::DbWeight::get().reads(2))
 			.saturating_add(T::DbWeight::get().writes(2))
 	}
@@ -87,9 +86,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `227`
 		//  Estimated: `8799`
-		// Minimum execution time: 126_831_000 picoseconds.
-		Weight::from_parts(127_541_000, 0)
-			.saturating_add(Weight::from_parts(0, 8799))
+		// Minimum execution time: 77_000_000 picoseconds.
+		Weight::from_parts(78_000_000, 8799)
 			.saturating_add(T::DbWeight::get().reads(8))
 			.saturating_add(T::DbWeight::get().writes(4))
 	}
@@ -97,9 +95,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 1_800_000 picoseconds.
-		Weight::from_parts(1_940_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(1_000_000, 0)
 	}
 	/// Storage: `ParachainInfo::ParachainId` (r:1 w:0)
 	/// Proof: `ParachainInfo::ParachainId` (`max_values`: Some(1), `max_size`: Some(4), added: 499, mode: `MaxEncodedLen`)
@@ -117,9 +114,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `175`
 		//  Estimated: `6196`
-		// Minimum execution time: 132_561_000 picoseconds.
-		Weight::from_parts(133_421_000, 0)
-			.saturating_add(Weight::from_parts(0, 6196))
+		// Minimum execution time: 90_000_000 picoseconds.
+		Weight::from_parts(90_000_000, 6196)
 			.saturating_add(T::DbWeight::get().reads(7))
 			.saturating_add(T::DbWeight::get().writes(3))
 	}
@@ -127,9 +123,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 4_120_000 picoseconds.
-		Weight::from_parts(4_390_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 2_000_000 picoseconds.
+		Weight::from_parts(2_000_000, 0)
 	}
 	/// Storage: `System::Account` (r:1 w:1)
 	/// Proof: `System::Account` (`max_values`: None, `max_size`: Some(128), added: 2603, mode: `MaxEncodedLen`)
@@ -137,9 +132,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `52`
 		//  Estimated: `3593`
-		// Minimum execution time: 35_160_000 picoseconds.
-		Weight::from_parts(35_500_000, 0)
-			.saturating_add(Weight::from_parts(0, 3593))
+		// Minimum execution time: 21_000_000 picoseconds.
+		Weight::from_parts(22_000_000, 3593)
 			.saturating_add(T::DbWeight::get().reads(1))
 			.saturating_add(T::DbWeight::get().writes(1))
 	}
@@ -159,9 +153,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `126`
 		//  Estimated: `6196`
-		// Minimum execution time: 105_511_000 picoseconds.
-		Weight::from_parts(106_471_000, 0)
-			.saturating_add(Weight::from_parts(0, 6196))
+		// Minimum execution time: 61_000_000 picoseconds.
+		Weight::from_parts(64_000_000, 6196)
 			.saturating_add(T::DbWeight::get().reads(7))
 			.saturating_add(T::DbWeight::get().writes(3))
 	}
@@ -181,29 +174,30 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `74`
 		//  Estimated: `3593`
-		// Minimum execution time: 61_881_000 picoseconds.
-		Weight::from_parts(62_530_000, 0)
-			.saturating_add(Weight::from_parts(0, 3593))
+		// Minimum execution time: 45_000_000 picoseconds.
+		Weight::from_parts(46_000_000, 3593)
 			.saturating_add(T::DbWeight::get().reads(6))
 			.saturating_add(T::DbWeight::get().writes(2))
 	}
-	/// Storage: `System::Account` (r:1 w:1)
+	/// Storage: `System::Account` (r:2 w:2)
 	/// Proof: `System::Account` (`max_values`: None, `max_size`: Some(128), added: 2603, mode: `MaxEncodedLen`)
-	/// Storage: `Dmp::DeliveryFeeFactor` (r:1 w:0)
-	/// Proof: `Dmp::DeliveryFeeFactor` (`max_values`: None, `max_size`: None, mode: `Measured`)
-	/// Storage: `XcmPallet::SupportedVersion` (r:1 w:0)
-	/// Proof: `XcmPallet::SupportedVersion` (`max_values`: None, `max_size`: None, mode: `Measured`)
-	/// Storage: `Dmp::DownwardMessageQueues` (r:1 w:1)
-	/// Proof: `Dmp::DownwardMessageQueues` (`max_values`: None, `max_size`: None, mode: `Measured`)
-	/// Storage: `Dmp::DownwardMessageQueueHeads` (r:1 w:1)
-	/// Proof: `Dmp::DownwardMessageQueueHeads` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `ParachainInfo::ParachainId` (r:1 w:0)
+	/// Proof: `ParachainInfo::ParachainId` (`max_values`: Some(1), `max_size`: Some(4), added: 499, mode: `MaxEncodedLen`)
+	/// Storage: `ParachainSystem::UpwardDeliveryFeeFactor` (r:1 w:0)
+	/// Proof: `ParachainSystem::UpwardDeliveryFeeFactor` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Storage: `PolkadotXcm::SupportedVersion` (r:1 w:0)
+	/// Proof: `PolkadotXcm::SupportedVersion` (`max_values`: None, `max_size`: None, mode: `Measured`)
+	/// Storage: `ParachainSystem::HostConfiguration` (r:1 w:0)
+	/// Proof: `ParachainSystem::HostConfiguration` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
+	/// Storage: `ParachainSystem::PendingUpwardMessages` (r:1 w:1)
+	/// Proof: `ParachainSystem::PendingUpwardMessages` (`max_values`: Some(1), `max_size`: None, mode: `Measured`)
 	pub(crate) fn initiate_transfer() -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `180`
-		//  Estimated: `3645`
-		// Minimum execution time: 82_584_000 picoseconds.
-		Weight::from_parts(84_614_000, 3645)
-			.saturating_add(T::DbWeight::get().reads(5))
+		//  Measured:  `126`
+		//  Estimated: `6196`
+		// Minimum execution time: 66_000_000 picoseconds.
+		Weight::from_parts(68_000_000, 6196)
+			.saturating_add(T::DbWeight::get().reads(7))
 			.saturating_add(T::DbWeight::get().writes(3))
 	}
 }
diff --git a/system-parachains/asset-hubs/asset-hub-polkadot/src/weights/xcm/pallet_xcm_benchmarks_generic.rs b/system-parachains/asset-hubs/asset-hub-polkadot/src/weights/xcm/pallet_xcm_benchmarks_generic.rs
index 2fcead851..70ea76a73 100644
--- a/system-parachains/asset-hubs/asset-hub-polkadot/src/weights/xcm/pallet_xcm_benchmarks_generic.rs
+++ b/system-parachains/asset-hubs/asset-hub-polkadot/src/weights/xcm/pallet_xcm_benchmarks_generic.rs
@@ -16,11 +16,11 @@

 //! Autogenerated weights for `pallet_xcm_benchmarks::generic`
 //!
-//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 47.0.0
-//! DATE: 2025-05-01, STEPS: `50`, REPEAT: `20`, LOW RANGE: `[]`, HIGH RANGE: `[]`
+//! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 43.0.0
+//! DATE: 2025-05-20, STEPS: `50`, REPEAT: `20`, LOW RANGE: `[]`, HIGH RANGE: `[]`
 //! WORST CASE MAP SIZE: `1000000`
-//! HOSTNAME: `0e68fc30b8dd`, CPU: `QEMU Virtual CPU version 2.5+`
-//! WASM-EXECUTION: `Compiled`, CHAIN: `None`, DB CACHE: 1024
+//! HOSTNAME: `mac-4.home`, CPU: `<UNKNOWN>`
+//! WASM-EXECUTION: Compiled, CHAIN: None, DB CACHE: 1024

 // Executed Command:
 // frame-omni-bencher
@@ -30,17 +30,17 @@
 // --extrinsic=*
 // --runtime=target/production/wbuild/asset-hub-polkadot-runtime/asset_hub_polkadot_runtime.wasm
 // --pallet=pallet_xcm_benchmarks::generic
-// --header=/_work/fellowship-001/runtimes/runtimes/.github/scripts/cmd/file_header.txt
+// --header=/Users/kkdev/Dev/karolk91/runtimes/.github/scripts/cmd/file_header.txt
 // --output=./system-parachains/asset-hubs/asset-hub-polkadot/src/weights/xcm
 // --wasm-execution=compiled
 // --steps=50
 // --repeat=20
 // --heap-pages=4096
+// --template=templates/xcm-bench-template.hbs

 #![cfg_attr(rustfmt, rustfmt_skip)]
 #![allow(unused_parens)]
 #![allow(unused_imports)]
-#![allow(missing_docs)]

 use frame_support::{traits::Get, weights::Weight};
 use core::marker::PhantomData;
@@ -64,9 +64,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `175`
 		//  Estimated: `6196`
-		// Minimum execution time: 107_232_000 picoseconds.
-		Weight::from_parts(135_321_000, 0)
-			.saturating_add(Weight::from_parts(0, 6196))
+		// Minimum execution time: 85_000_000 picoseconds.
+		Weight::from_parts(92_000_000, 6196)
 			.saturating_add(T::DbWeight::get().reads(7))
 			.saturating_add(T::DbWeight::get().writes(3))
 	}
@@ -74,9 +73,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 690_000 picoseconds.
-		Weight::from_parts(1_050_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(1_000_000, 0)
 	}
 	/// Storage: `System::Account` (r:1 w:1)
 	/// Proof: `System::Account` (`max_values`: None, `max_size`: Some(128), added: 2603, mode: `MaxEncodedLen`)
@@ -84,9 +82,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `3593`
-		// Minimum execution time: 6_590_000 picoseconds.
-		Weight::from_parts(8_950_000, 0)
-			.saturating_add(Weight::from_parts(0, 3593))
+		// Minimum execution time: 5_000_000 picoseconds.
+		Weight::from_parts(6_000_000, 3593)
 			.saturating_add(T::DbWeight::get().reads(1))
 			.saturating_add(T::DbWeight::get().writes(1))
 	}
@@ -94,9 +91,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 760_000 picoseconds.
-		Weight::from_parts(1_000_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(0, 0)
 	}
 	/// Storage: `PolkadotXcm::Queries` (r:1 w:0)
 	/// Proof: `PolkadotXcm::Queries` (`max_values`: None, `max_size`: None, mode: `Measured`)
@@ -104,58 +100,51 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `3465`
-		// Minimum execution time: 6_190_000 picoseconds.
-		Weight::from_parts(7_022_000, 0)
-			.saturating_add(Weight::from_parts(0, 3465))
+		// Minimum execution time: 4_000_000 picoseconds.
+		Weight::from_parts(4_000_000, 3465)
 			.saturating_add(T::DbWeight::get().reads(1))
 	}
 	pub(crate) fn transact() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 7_260_000 picoseconds.
-		Weight::from_parts(8_830_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 4_000_000 picoseconds.
+		Weight::from_parts(5_000_000, 0)
 	}
 	pub(crate) fn refund_surplus() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 2_730_000 picoseconds.
-		Weight::from_parts(2_921_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 1_000_000 picoseconds.
+		Weight::from_parts(2_000_000, 0)
 	}
 	pub(crate) fn set_error_handler() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 680_000 picoseconds.
-		Weight::from_parts(780_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(1_000_000, 0)
 	}
 	pub(crate) fn set_appendix() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 660_000 picoseconds.
-		Weight::from_parts(770_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(1_000_000, 0)
 	}
 	pub(crate) fn clear_error() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 700_000 picoseconds.
-		Weight::from_parts(820_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(0, 0)
 	}
 	pub(crate) fn descend_origin() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 730_000 picoseconds.
-		Weight::from_parts(790_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(0, 0)
 	}
 	/// Storage: `Benchmark::Override` (r:0 w:0)
 	/// Proof: `Benchmark::Override` (`max_values`: None, `max_size`: None, mode: `Measured`)
@@ -165,15 +154,13 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		//  Estimated: `0`
 		// Minimum execution time: 18_446_744_073_709_551_000 picoseconds.
 		Weight::from_parts(18_446_744_073_709_551_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
 	}
 	pub(crate) fn clear_origin() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 720_000 picoseconds.
-		Weight::from_parts(850_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(1_000_000, 0)
 	}
 	/// Storage: `ParachainInfo::ParachainId` (r:1 w:0)
 	/// Proof: `ParachainInfo::ParachainId` (`max_values`: Some(1), `max_size`: Some(4), added: 499, mode: `MaxEncodedLen`)
@@ -191,9 +178,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `175`
 		//  Estimated: `6196`
-		// Minimum execution time: 71_260_000 picoseconds.
-		Weight::from_parts(78_911_000, 0)
-			.saturating_add(Weight::from_parts(0, 6196))
+		// Minimum execution time: 48_000_000 picoseconds.
+		Weight::from_parts(50_000_000, 6196)
 			.saturating_add(T::DbWeight::get().reads(7))
 			.saturating_add(T::DbWeight::get().writes(3))
 	}
@@ -203,9 +189,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `23`
 		//  Estimated: `3488`
-		// Minimum execution time: 9_680_000 picoseconds.
-		Weight::from_parts(11_131_000, 0)
-			.saturating_add(Weight::from_parts(0, 3488))
+		// Minimum execution time: 6_000_000 picoseconds.
+		Weight::from_parts(7_000_000, 3488)
 			.saturating_add(T::DbWeight::get().reads(1))
 			.saturating_add(T::DbWeight::get().writes(1))
 	}
@@ -213,9 +198,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 669_000 picoseconds.
-		Weight::from_parts(840_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(0, 0)
 	}
 	/// Storage: `PolkadotXcm::VersionNotifyTargets` (r:1 w:1)
 	/// Proof: `PolkadotXcm::VersionNotifyTargets` (`max_values`: None, `max_size`: None, mode: `Measured`)
@@ -231,9 +215,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `42`
 		//  Estimated: `3507`
-		// Minimum execution time: 22_670_000 picoseconds.
-		Weight::from_parts(33_480_000, 0)
-			.saturating_add(Weight::from_parts(0, 3507))
+		// Minimum execution time: 16_000_000 picoseconds.
+		Weight::from_parts(17_000_000, 3507)
 			.saturating_add(T::DbWeight::get().reads(5))
 			.saturating_add(T::DbWeight::get().writes(2))
 	}
@@ -243,50 +226,44 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 3_470_000 picoseconds.
-		Weight::from_parts(4_070_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 2_000_000 picoseconds.
+		Weight::from_parts(2_000_000, 0)
 			.saturating_add(T::DbWeight::get().writes(1))
 	}
 	pub(crate) fn burn_asset() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 24_720_000 picoseconds.
-		Weight::from_parts(25_170_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 33_000_000 picoseconds.
+		Weight::from_parts(33_000_000, 0)
 	}
 	pub(crate) fn expect_asset() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 7_299_000 picoseconds.
-		Weight::from_parts(7_470_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 15_000_000 picoseconds.
+		Weight::from_parts(16_000_000, 0)
 	}
 	pub(crate) fn expect_origin() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 640_000 picoseconds.
-		Weight::from_parts(710_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(1_000_000, 0)
 	}
 	pub(crate) fn expect_error() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 660_000 picoseconds.
-		Weight::from_parts(760_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(1_000_000, 0)
 	}
 	pub(crate) fn expect_transact_status() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 849_000 picoseconds.
-		Weight::from_parts(960_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(0, 0)
 	}
 	/// Storage: `ParachainInfo::ParachainId` (r:1 w:0)
 	/// Proof: `ParachainInfo::ParachainId` (`max_values`: Some(1), `max_size`: Some(4), added: 499, mode: `MaxEncodedLen`)
@@ -304,9 +281,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `175`
 		//  Estimated: `6196`
-		// Minimum execution time: 75_601_000 picoseconds.
-		Weight::from_parts(79_281_000, 0)
-			.saturating_add(Weight::from_parts(0, 6196))
+		// Minimum execution time: 51_000_000 picoseconds.
+		Weight::from_parts(52_000_000, 6196)
 			.saturating_add(T::DbWeight::get().reads(7))
 			.saturating_add(T::DbWeight::get().writes(3))
 	}
@@ -314,9 +290,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 5_170_000 picoseconds.
-		Weight::from_parts(5_280_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 3_000_000 picoseconds.
+		Weight::from_parts(3_000_000, 0)
 	}
 	/// Storage: `ParachainInfo::ParachainId` (r:1 w:0)
 	/// Proof: `ParachainInfo::ParachainId` (`max_values`: Some(1), `max_size`: Some(4), added: 499, mode: `MaxEncodedLen`)
@@ -334,9 +309,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `175`
 		//  Estimated: `6196`
-		// Minimum execution time: 70_280_000 picoseconds.
-		Weight::from_parts(73_230_000, 0)
-			.saturating_add(Weight::from_parts(0, 6196))
+		// Minimum execution time: 48_000_000 picoseconds.
+		Weight::from_parts(49_000_000, 6196)
 			.saturating_add(T::DbWeight::get().reads(7))
 			.saturating_add(T::DbWeight::get().writes(3))
 	}
@@ -344,25 +318,22 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 740_000 picoseconds.
-		Weight::from_parts(811_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(1_000_000, 0)
 	}
 	pub(crate) fn set_topic() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 629_000 picoseconds.
-		Weight::from_parts(720_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(1_000_000, 0)
 	}
 	pub(crate) fn clear_topic() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 671_000 picoseconds.
-		Weight::from_parts(730_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(0, 0)
 	}
 	/// Storage: `System::Account` (r:1 w:1)
 	/// Proof: `System::Account` (`max_values`: None, `max_size`: Some(128), added: 2603, mode: `MaxEncodedLen`)
@@ -374,9 +345,8 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `491`
 		//  Estimated: `4273`
-		// Minimum execution time: 79_271_000 picoseconds.
-		Weight::from_parts(85_471_000, 0)
-			.saturating_add(Weight::from_parts(0, 4273))
+		// Minimum execution time: 54_000_000 picoseconds.
+		Weight::from_parts(54_000_000, 4273)
 			.saturating_add(T::DbWeight::get().reads(3))
 			.saturating_add(T::DbWeight::get().writes(3))
 	}
@@ -386,33 +356,29 @@ impl<T: frame_system::Config> WeightInfo<T> {
 		// Proof Size summary in bytes:
 		//  Measured:  `32`
 		//  Estimated: `1489`
-		// Minimum execution time: 4_200_000 picoseconds.
-		Weight::from_parts(4_390_000, 0)
-			.saturating_add(Weight::from_parts(0, 1489))
+		// Minimum execution time: 3_000_000 picoseconds.
+		Weight::from_parts(4_000_000, 1489)
 			.saturating_add(T::DbWeight::get().reads(1))
 	}
 	pub(crate) fn set_fees_mode() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 649_000 picoseconds.
-		Weight::from_parts(720_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(0, 0)
 	}
 	pub(crate) fn unpaid_execution() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 660_000 picoseconds.
-		Weight::from_parts(740_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(1_000_000, 0)
 	}
 	pub(crate) fn alias_origin() -> Weight {
 		// Proof Size summary in bytes:
 		//  Measured:  `0`
 		//  Estimated: `0`
-		// Minimum execution time: 649_000 picoseconds.
-		Weight::from_parts(711_000, 0)
-			.saturating_add(Weight::from_parts(0, 0))
+		// Minimum execution time: 0_000 picoseconds.
+		Weight::from_parts(0, 0)
 	}
 }
 ```

bkontur commented:
> Since its not really possible to test this changes using comment bot (the bot will run using code from `master`), for review purposes, please find `git diff` of my local example run of
> 
> ```shell
> python3 .github/scripts/cmd/cmd.py bench --runtime asset-hub-polkadot --pallet pallet_xcm_benchmarks::generic pallet_xcm_benchmarks::fungible pallet_xcm
> ```


nice, looks like it works for `pallet_xcm_benchmarks::generic pallet_xcm_benchmarks::fungible`

bkontur commented:
> Since its not really possible to test this changes using comment bot (the bot will run using code from `master`), for review purposes, please find `git diff` of my local example run of
> 
> ```shell
> python3 .github/scripts/cmd/cmd.py bench --runtime asset-hub-polkadot --pallet pallet_xcm_benchmarks::generic pallet_xcm_benchmarks::fungible pallet_xcm
> ```

@karolk91 Could you please update https://github.com/polkadot-fellows/runtimes/blob/main/docs/weight-generation.md in a separate PR? Specifically, please add clear steps on how to run the `cmd-bot` locally - for example, for cases where we run it on a dedicated machine. This way, we won’t need to keep creating custom scripts to repeatedly handle things like the XCM benchmarks subdirectory hack - this will be less error prone. Ideally, the processing and the resulting output should be exactly the same as:
- when we use cmd bot in the PR
- when we use cmd bot in the dedicated machine or locally

Ideally:
```
# prerequisites
# Installed python

git clone ...
<TODO> example of how to run cmd bot
```


cc: @ggwpez
